server.js

```diff
@@ -1,28 +1,85 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import listEndpoints from 'express-list-endpoints'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndexes: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    trim: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: () => Date.now
+  }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((_, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service not available' })
+  }
+})
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (_, res) => {
+  res.send(listEndpoints(app))
 })
 
+app.get('/thoughts', async (_, res) => {
+  try {
+    const everyThought = await Thought.find().sort({ createdAt: -1 }).limit(20)
+    res.json(everyThought)
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought(req.body).save()
+    res.json(newThought)
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const newLike = await Thought.findByIdAndUpdate({ id }, {$inc: {hearts: 1} })
```

Comment:
It is not possible to add likes right now, I checked it in Postman and it does not work there, so there is something wrong in the backend. I believe that you only need to write id on line 73 and not { id } :)